### PR TITLE
Webhook notifications support (WIP)

### LINKED ISSUE #72: Back : Add Webhook Events
Hi
Please add the Webhook Events feature as well. To send a request to an address in the following cases.

Enable / Disable User
To finish traffic or user time
Change time or user traffic

thank you

### PR DESCRIPTION
**Pull request by govfvck**
fixes #72

Implements webhook notifications!
Events on a user will be sent to a webhook address provided by the `WEBHOOK_ADDRESS` config variable.

also updated readme for more info

this pull request is a work in progress and needs some review. things might change!

### CODE REVIEW

File: app/jobs/send_notifications.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            if ((dt.utcnow().timestamp() - notification.enqueued_at) > 60 * 60 * 4) or notification.tries > 3:
+                # ignore notifications after 4 hours or 3 retries
+                continue
+            if not send(notification.dict()):
+                notification.tries += 1
+                queue.append(notification)
+                return  # stop sending notifications temporarily
+    except IndexError:
+        pass
+
```
Comment (SaintShit) [inline app/jobs/send_notifications.py:39]: Please add a event handler that runs send_notifications on shutdown event before shutdown.

**Review by SaintShit — CHANGES_REQUESTED** [review]
Please add a event handler that runs send_notifications on shutdown event before shutdown.

**Comment by SaintShit** [comment]
thanks.

**Comment by AC-Lover** [comment]
I tested it, unfortunately it didn't work at all =(
Please check and fix, thank you

**Comment by AC-Lover** [comment]
Hello
Why does it not work at all on `https`?
Must be `http` to work
Please check, I think it has a bug

**Pull request merged**